lib/puppet/functions/stdlib/ssl/validate_key_pair.rb

```diff
@@ -0,0 +1,27 @@
+require 'openssl'
+# @summary
+#    Validates a PEM-formatted X.509 public_key and private key using
+#    OpenSSL. Verifies that the public_key's matches the supplied key.
+Puppet::Functions.create_function(:'stdlib::ssl::validate_key_pair') do
+  dispatch :validate_ssl_key_pair do
+    param 'String[1]', :public_key
+    param 'String[1]', :private_key
+    return_type 'Boolean'
```

Comment:
Maybe don't return anything.  All the other `validate_` functions fail or return nothing.

Reply:
hmm tbh i didn't like using the validate_ name as it makes me think of type validators before we had custom types, but i couldn't think of a better name. Naming aside  i think returning a boolean is a bit nicer; as an contrived example of the top of my head; you could have a pem file with the full chain and want to loop through and make sure at least one match the private key.